src/diagonal.works/b6/world.go

```diff
@@ -936,80 +1060,102 @@ func Centroid(geometry Geometry) s2.Point {
 	return s2.Point{}
 }
 
-type Geo struct {
-	ll s2.LatLng
+func GeometryToGeoJSON(g Geometry) *geojson.Feature {
+	switch g.GeometryType() {
+	case GeometryTypePoint:
+		return geojson.NewFeatureFromS2Point(g.Point())
+	case GeometryTypePath:
+		return geojson.NewFeatureWithGeometry(geojson.GeometryFromLineString(geojson.FromPolyline(g.Polyline())))
+	default:
+		panic("not implemented")
+	}
 }
 
-func (g Geo) GeometryType() GeometryType {
-	return GeometryTypePoint
+type Feature interface {
+	Identifiable
+	Taggable
+	References() []Reference
+	Reference(i int) Reference
 }
 
-func (g Geo) Location() s2.LatLng {
-	return g.ll
+type PhysicalFeature interface {
+	Feature
+	Geometry
 }
 
-func (g Geo) ToGeoJSON() geojson.GeoJSON {
-	return LocationToGeoJSON(g)
+type NestedPhysicalFeature interface { // m: rename
```

Comment:
Seems odd to call this one out as a separate interface, when the Geometry interface already contains the union of all methods (some of which will be invalid based on the actual underlying geometry type). It feels to me that they should mirror each other?

Reply:
ok i m leaving a todo here, just so i can submit smth / and i ll pick it up again first thing

i need to get rid of having this be physical (&& referencing physical) which is just a bit of a pain due to segments but it s doable and i deffo need a nested thing to overwrite geometry (b/c references can point to actual features / and so in some cases there could or could not be points present) but yes i feel like it can be done more subtly / and be part of a feature interface rather than a separate one (i m leaning towards this but started doing it and it s becoming a bit big)